### PR TITLE
nixos/unifi-video: add deprecation warning for openFirewall

### PR DESCRIPTION
**Pull request by rsynnest**
Follows example of https://github.com/NixOS/nixpkgs/pull/150248 to provide a deprecation notice for openFirewall default = true  as requested in #149889

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by pennae — CHANGES_REQUESTED** [review]
it doesn't quite follow the example: renaming the option should be done with mkRenamedOption to get a warning about the renaming instead of a build error, and a changelog entry for this change in behaviour is also warranted.

**Comment by rsynnest** [comment]
Thank you @pennae ! I've added those changes as well as some formatting and a bugfix for int -> string conversion (hope it's ok in the one PR?).

File: nixos/modules/services/video/unifi-video.nix [hunk 1]
```diff
@@ -91,98 +91,102 @@ let
...
+      (options.services.unifi-video.openFirewall.highestPrio >= (mkOptionDefault null).priority)
+      "The current services.unifi-video.openFirewall = true default is deprecated and will change to false in 22.11. Set it explicitly to silence this warning.";
 
-config = mkIf cfg.enable {
-  users = {
-    users.unifi-video = {
+    users.users.unifi-video = {
       description = "UniFi Video controller daemon user";
-      home = stateDir;
+      home = "${stateDir}";
```
Comment (pennae) [inline nixos/modules/services/video/unifi-video.nix:183]: that could've stayed how it was :)

**Review by pennae — COMMENTED** [review]
changes look fine, but now the manual check fails (looks like a missed run of md-to-db). just found one completely inconsequential thing looking through.

File: nixos/modules/services/video/unifi-video.nix (hunk 1 above)
Comment (rsynnest) [reply nixos/modules/services/video/unifi-video.nix]: ah ok, wasn't sure what the convention is or if previous one was less safe for spaces or something. Changed it back and re-pushed, maybe the manual check will pass this time too.

**Comment by pennae** [comment]
manual check still fails :( (did you write the docbook xml by hand?)

**Comment by rsynnest** [comment]
Oh oops yes, sorry about that, this is my first time changing the manual. Removed the docbook edit and tested locally this time with `md-to-db.sh` + nix-build.

**Comment by pennae** [comment]
eh, mistakes happen :) the files `md-to-db` generates still have to be commited though (that's what the manual check checks: that md-to-db doesn't produce any changes). after that this should be fine.

**Pull request merged**